src/Collection/CollectionInterface.php

```diff
@@ -832,6 +849,44 @@ public function unfold(callable $transformer = null);
     public function through(callable $handler);
 
     /**
+     * Combines the elements of this collection with each of the elements of the
+     * passed iterables, using their positional index as a reference.
+     *
+     * ### Example:
+     *
+     * ```
+     * $collection = new Collection([1, 2]);
+     * $collection->zip([3, 4], [5, 6])->toList(); // returns [[1, 3, 5], [2, 4, 6]]
+     * ```
+     *
+     * @param array|\Traversable ...$items The collections to zip.
```

Comment:
Yes, I'm starting to use the php 5.6 way of hinting variable arguments of the same type